apps/stripe/src/app/api/saleor/transaction-initialize-session/use-case.ts

```diff
@@ -24,6 +31,66 @@ export class TransactionInitializeSessionUseCase {
     this.stripePaymentIntentsApiFactory = deps.stripePaymentIntentsApiFactory;
   }
 
+  private prepareStripeCreatePaymentIntentParams(
+    event: TransactionInitializeSessionEventFragment,
+  ): Stripe.PaymentIntentCreateParams {
+    const stripeMoneyResult = StripeMoney.createFromSaleorAmount({
+      amount: event.action.amount,
+      currency: event.action.currency,
+    });
+
+    if (stripeMoneyResult.isErr()) {
+      this.logger.error(
+        "Failed to convert amount / currency coming from Saleor to one used by Stripe - throwing",
+        {
+          error: stripeMoneyResult.error,
+        },
+      );
+
+      throw new TransactionInitializeSessionUseCase.UseCaseError("Failed to create Stripe money", {
+        cause: stripeMoneyResult.error,
+      });
+    }
+
+    return {
+      amount: stripeMoneyResult.value.getAmount(),
+      currency: stripeMoneyResult.value.getCurrency(),
+    };
+  }
+
+  private prepareResponseToSaleor(
+    stripePaymentIntentResponse: Stripe.PaymentIntent,
+  ): ReturnType<typeof buildSyncWebhookResponsePayload<"TRANSACTION_INITIALIZE_SESSION">> {
```

Comment:
This type is having "data" as unknown. Why don't you:
```
type ValidResponse = SyncWebhookResponsesMap["TRANSACTION_INITIALIZE_SESSION"] & {
  data : {
    stripeClientSecret: string | null;
  }
}
```

So we have single place how response looks like.
(and this method signature is too loose)

Reply:
Done in https://github.com/saleor/apps/pull/1851/commits/70500c9777ef174b2814e501adb64d90e11be615